include/uri/UResource.h

```diff
@@ -0,0 +1,292 @@
+     * @param message The Message type matches the protobuf service IDL message name that defines structured data types.
+     *                A message is a data structure type used to define data that is passed in  events and rpc methods.
+     * @return Returns a UResource that can be serialised into a long UUri.
+     */
+    static UResource longFormat(const std::string& name,
+                                const std::string& instance,
+                                const std::string& message) {
+        return UResource{name, instance, message, std::nullopt, false};
+    }
+
+    /**
+     * Build a UResource that can be serialised into a micro UUri. Mostly used for publishing messages.
+     * @param id The numeric representation of this uResource.
+     * @return Returns a UResource that can be serialised into a micro UUri.
+     */
+    static UResource microFormat(const std::optional<uint16_t> id) {
+        return UResource{"", "", "", id, false};
+    }
+
+    /**
+     * Build a UResource for rpc request, using only the long format.
+     * @param methodName The RPC method name.
+     * @return Returns a UResource used for an RPC request that could be serialised in long format.
+     */
+    static UResource forRpcRequest(const std::string& methodName) {
+        return UResource{"rpc", methodName, "", std::nullopt, false};
+    }
+
+    /**
+     * Build a UResource for rpc request, using only the micro format.
+     * @param methodId The numeric representation method name for the RPC.
+     * @return Returns a UResource used for an RPC request that could be serialised in micro format.
+     */
+    static UResource forRpcRequest(const std::optional<uint16_t> methodId) {
+        return UResource{"rpc", "", "", methodId, false};
+    }
+
+    /**
+     * Build a UResource for rpc request, using both the long and micro format information.
+     * @param methodName The RPC method name.
+     * @param methodId The numeric representation method name for the RPC.
+     * @return Returns a UResource used for an RPC request that could be serialised in long and micro format.
+     */
+    static UResource forRpcRequest(const std::string& methodName,
+                                   const std::optional<uint16_t> methodId) {
+        bool resolved = !isBlank(methodName) && methodId.has_value();
+        return UResource{"rpc", methodName, "", methodId, resolved};
+    }
+
+    /**
+     * Static factory method for creating a response resource that is returned from RPC calls<br>
+     * @return Returns a response  resource used for response RPC calls.
+     */
+    static UResource forRpcResponse() {
+        static const auto RESPONSE = UResource("rpc", "response", "", 0, true);
+        return RESPONSE;
+    }
+
+    /**
+     * @return Returns true if this resource specifies an RPC method call or RPC response.
+     */
+    [[nodiscard]] bool isRPCMethod() const {
```

Comment:
can you please confirm that it works for MICRO uri ?

Reply:
It works for micro URI. Unit test cases are available.